src/scenechange/mod.rs

```diff
@@ -239,6 +237,14 @@ impl<T: Pixel> SceneChangeDetector<T> {
     let mut cloned_deque = self.score_deque.to_vec();
     cloned_deque.remove(self.deque_offset);
 
+
+    // Subtract the previous metric value from the current one
+    // It makes the peaks in the metric more distinctive
```

Comment:
I might like a bit more analysis of this. In particular, this effectively limits us to a scene cut only every other frame, correct?

Reply:
The visualization you can see at the end of this [post](https://dev.to/aleksandrgushchin/threshold-experiments-for-scene-change-detector-2hbp). The new metric can be interpreted as the rate of change in dissimilarity metric between successive frames. Since the action inside the scene is usually monotonous, the dissimilarity metric stays about the same or changes at a small rate. However, there are scenes whose movement within a frame causes the dissimilarity metric to have high values, even exceeding the values of some scene changes.
The new metric solves this problem by decreasing the values on such scenes. At the same time dissimilarity metric peaks on frames with scene change remain high enough for detection with the threshold.

Reply:
Great, OK it does look like a big improvement.

Have you done a run on AWCY with vimeo-corpus-10s? I can help you do this if you like.

What is the intended behavior for the second frame in a scene, when the previous scene's score isn't available? This looks like it treats the previous frame as having a delta of 0. This seems like it'd effectively make a scenecut here more likely than average always.

Have you considered doing a delta of an average of x previous frames?

Reply:
> Have you done a run on AWCY with vimeo-corpus-10s? I can help you do this if you like.

Not yet, it would be helpful!

> What is the intended behavior for the second frame in a scene, when the previous scene's score isn't available? 

On the first few frames of the video the previous metric has low values, so subtracting one frame metric value from the other should not exceed the threshold. 

> Have you considered doing a delta of an average of x previous frames?

Not yet, because this new metric solved my problem, but I'll do experiments with it.